skyportal/handlers/api/gcn.py

```diff
@@ -660,3 +671,491 @@ def delete(self, dateobs, localization_name):
+            from_str = f"""FROM:  {self.associated_user_object.first_name} {self.associated_user_object.last_name} at Affiliation"""
+            if self.associated_user_object.contact_email is not None:
+                from_str += f""" <{self.associated_user_object.contact_email}>\n"""
+            header_text.append(from_str)
+
+            if len(user_ids) > 0:
+                # query user objects for all user_ids
+                users = []
+                for user_id in user_ids:
+                    user = User.query.get(user_id)
+                    if user is None:
+                        return self.error(f"User ID {user_id} not found")
+                    users.append(user)
+
+                users_txt = []
+                for user in users:
+                    if user.first_name is not None and user.last_name is not None:
+                        users_txt.append(
+                            f"""{user.first_name[0].upper()}. {user.last_name} (Affiliation)"""  # hardcoded affiliation as it is not implemented yet
+                        )
+                # create a string of all users, with 5 users per line
+                users_txt = "\n".join(
+                    [
+                        ", ".join(users_txt[i : i + 5])
+                        for i in range(0, len(users_txt), 5)
+                    ]
+                )
+                header_text.append(f"""\n{users_txt}\n""")
+
+            header_text.append(f"""\non behalf of the {group.name}, report:\n""")
+            contents.extend(header_text)
+
+        if show_sources:
+            sources_text = []
+            source_page_number = 1
+            source_num_per_page = 100
+            sources = []
+            while True:
+                # get the sources in the event
+                sources_data = get_sources(
+                    user_id=self.associated_user_object.id,
+                    session=DBSession(),
+                    first_detected_date=start_date,
+                    last_detected_date=end_date,
+                    localization_dateobs=dateobs,
+                    localization_name=localization_name,
+                    localization_cumprob=localization_cumprob,
+                    include_photometry=True,
+                    page_number=source_page_number,
+                    num_per_page=source_num_per_page,
+                    photometry_format='mag',
+                )
+                sources.extend(sources_data['sources'])
+                source_page_number += 1
+
+                if len(sources_data['sources']) < source_num_per_page:
+                    break
+            if len(sources) > 0:
+                sources_text.append(
+                    f"\nFound {len(sources)} {'sources' if len(sources) > 1 else 'source'} in the event's localization, given the specified date range:\n"
+                ) if not no_text else None
+                ids, aliases, ras, decs, ra_errs, dec_errs, redshifts = (
```

Comment:
I think we can do without the ra/dec uncertainties. redshift uncertainty if it is there though is nice.